src/shogun/kernel/CombinedKernel.cpp

```diff
@@ -121,8 +121,13 @@ bool CCombinedKernel::init(CFeatures* l, CFeatures* r)
 		// skip over features - the custom kernel does not need any
 		if (k->get_kernel_type() != K_CUSTOM)
 		{
-			lf = ((CCombinedFeatures*) l)->get_feature_obj(f_idx);
-			rf = ((CCombinedFeatures*) r)->get_feature_obj(f_idx);
+			if (((CCombinedFeatures*)l)->get_num_feature_obj() > f_idx &&
```

Comment:
Same thing as above, I tried preserving as much of the original code as possible (perhaps not the best choice). The original code seems to expect 0 being returned by convention if index is out of bounds when accessing data from multiple structures. However if you agree i'd be more than happy to rework it so that this isnt the case any more.